compiler/src/toEC.ml

```diff
@@ -468,39 +470,22 @@ let get_funtype env f = snd (Mf.find f env.funs)
 
 let get_funname env f = fst (Mf.find f env.funs) 
 
-let get_aux ?(fresh = false) env prefix tys =
-  let do1 ty =
-    let i = try Mty.find (prefix, ty) !(env.count) with Not_found -> 0 in
-    let l = try Mty.find (prefix, ty) !(env.auxv) with Not_found -> [] in
-    if (not fresh) && 0 < List.length l then
-      List.last l
-    else
-      begin
-        if i < List.length l then
-          begin
-            env.count := Mty.add (prefix,ty) (i+1) !(env.count);
-            List.nth l i
-          end
-        else
-          let aux = create_name env prefix in
-          env.auxv := Mty.add (prefix, ty) (l @ [aux]) !(env.auxv);
-          env.alls := Ss.add aux !(env.alls);
-          env.count := Mty.add (prefix, ty) (i+1) !(env.count);
-          aux
-      end
-  in
-  List.map do1 tys
-
-let pop_aux env prefix ty =
+let get_aux ?(fresh = true) env prefix ty =
   let i = try Mty.find (prefix, ty) !(env.count) with Not_found -> 0 in
-  if 0 < i then
-    env.count := Mty.add (prefix,ty) (i-1) !(env.count)
-
-let reset_aux env prefix tys =
-  let do1 ty =
-    env.count:= Mty.add (prefix, ty) 0 !(env.count)
-  in
-  List.iter do1 tys
+  let l = try Mty.find (prefix, ty) !(env.auxv) with Not_found -> [] in
+  env.count := Mty.add (prefix,ty) (i+1) !(env.count);
```

Comment:
I think increasing the counter of the aux variables should be done only when `fresh` is `true`.

Reply:
indeed